### PR TITLE
Fixups

### PR DESCRIPTION
**Pull request by bestie**
Rubocop was failing with many complaints, this fixes them.

* Mostly automated fixes (first commit)
* Manual code tweaks for complex fixes (later commits)
* Tests pass on native machine as well in Docker (last commit)

### CODE REVIEW

File: ruby/example_code.rb [hunk 1]
```diff
@@ -7,20 +7,17 @@
...
-                             :output => $stdout,
-                             :events => MY_EVENTS
-                            )
+log = MicroserviceLogger.new(service_name: 'my_super_service',
+                             clock: Time,
+                             output: $stdout,
+                             events: MY_EVENTS)
 
-log.startup.error({ message: "Emergency! There's an Emergency going on" })
+log.startup.error(message: "Emergency! There's an Emergency going on")
```
Comment (sleepyfox) [inline ruby/example_code.rb]: Is 'implicit object' really a rubocop style default?

File: ruby/example_code.rb [hunk 2]
```diff
@@ -7,20 +7,17 @@
...
   :my_event,
   :my_other_event
-]
+].freeze
 
-log = MicroserviceLogger.new(:service_name => 'my_super_service',
-                             :clock => Time,
-                             :output => $stdout,
-                             :events => MY_EVENTS
-                            )
```
Comment (sleepyfox) [inline ruby/example_code.rb]: I thought :symbol => 'value' was the preferred form for hashes in Ruby 2.x?

File: ruby/lib/microservice_logging.rb [hunk 3]
```diff
@@ -40,10 +34,28 @@ def method_missing(event)
...
+                           clock: @clock,
+                           output: @output,
+                           scoped_properties: scoped_properties)
+  end
+
+  private
+
+  def event_loggers
+    @event_loggers ||= Hash[
+      @events.map do |method_name|
```
Comment (sleepyfox) [inline ruby/lib/microservice_logging.rb]: I'm unclear what this is supposed to be fixing?

File: ruby/lib/microservice_logging.rb [hunk 4]
```diff
@@ -14,22 +14,16 @@ def initialize(service_name:,
...
 
-  def method_missing(event)
-    super unless @events.include? event
-    @event_loggers[event]
+  def method_missing(event_or_method_name)
+    super unless respond_to_missing?(event_or_method_name)
+    event_loggers.fetch(event_or_method_name)
+  end
+
+  def respond_to_missing?(event_or_method_name)
```
Comment (sleepyfox) [inline ruby/lib/microservice_logging.rb]: Why does adding an extra method for a what is already a 2-line method make things more readable?

File: ruby/lib/microservice_logging.rb (hunk 3 above)
Comment (bestie) [inline ruby/lib/microservice_logging.rb]: Rubocop issue with this was that the constructor was too long, hence extracting to a method. Once the method is extracted is creating the hash lazily is just more idiomatic.

File: ruby/lib/microservice_logging.rb (hunk 4 above)
Comment (bestie) [inline ruby/lib/microservice_logging.rb]: Ruby objects should know what methods they respond to, adding this allows the `#respond_to?` to return the correct answer.

File: ruby/example_code.rb (hunk 1 above)
Comment (bestie) [inline ruby/example_code.rb]: For method arguments yes.

File: ruby/example_code.rb (hunk 2 above)
Comment (bestie) [inline ruby/example_code.rb]: Sadly I don't see a lot of hash rockets these days, the Rubocop default is definitely in line with common practice in this case.

Comment (sleepyfox) [inline ruby/example_code.rb]: So sad.

File: ruby/example_code.rb (hunk 1 above)
Comment (sleepyfox) [inline ruby/example_code.rb]: OK...

File: ruby/lib/microservice_logging.rb (hunk 4 above)
Comment (sleepyfox) [inline ruby/lib/microservice_logging.rb]: OK.

File: ruby/lib/microservice_logging.rb (hunk 3 above)
Comment (sleepyfox) [inline ruby/lib/microservice_logging.rb]: OK.

**Pull request merged**